src/LendingPool.sol

```diff
@@ -801,6 +835,46 @@ contract LendingPool is LendingPoolGuardian, Creditor, DebtToken, InterestRateMo
      */
     function updateInterestRate() external processInterests { }
 
+    /**
+     * @notice Calculates the interest rate.
+     * @param utilisation Utilisation rate, 4 decimal precision.
+     * @return interestRate_ The current interest rate, 18 decimal precision.
+     * @dev The interest rate is a function of the utilisation of the Lending Pool.
+     * We use two linear curves: one below the optimal utilisation with low slope and a steep one above.
+     */
+    function _calculateInterestRate(uint256 utilisation) internal view returns (uint256 interestRate_) {
+        unchecked {
+            if (utilisation >= utilisationThreshold) {
+                // lsIR (1e22) = uT (1e4) * ls (1e18).
+                uint256 lowSlopeInterest = uint256(utilisationThreshold) * lowSlopePerYear;
+                // hsIR (1e22) = (u - uT) (1e4) * hs (e18).
+                uint256 highSlopeInterest = uint256(utilisation - utilisationThreshold) * highSlopePerYear;
+                // i (1e18) =  (lsIR (e22) + hsIR (1e22)) / 1e4 + bs (1e18).
+                interestRate_ = (lowSlopeInterest + highSlopeInterest) / ONE_4 + baseRatePerYear;
+            } else {
+                // i (1e18) = (u (1e4) * ls (1e18)) / 1e4 + br (1e18).
+                interestRate_ = utilisation * lowSlopePerYear / ONE_4 + baseRatePerYear;
+            }
+        }
+    }
+
+    /**
+     * @notice Updates the interest rate.
+     * @param totalDebt Total amount of debt.
+     * @param totalLiquidity Total amount of Liquidity (sum of borrowed out assets and assets still available in the Lending Pool).
+     */
+    function _updateInterestRate(uint256 totalDebt, uint256 totalLiquidity) internal {
```

Comment:
Change order _updateInterestRate and _calculateInterestRate.
Better to have view functions after ones that write to state.